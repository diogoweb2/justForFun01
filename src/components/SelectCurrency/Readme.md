```js
<SelectCurrency currencies={["MEX", "REL", "ZAB"]} />
```
