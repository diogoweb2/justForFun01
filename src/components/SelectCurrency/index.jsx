import React from "react";
import PropTypes from "prop-types";

const SelectCurrency = ({ currencies, onChange, value }) => (
  <select value={value} onChange={e => onChange(e.target.value)}>
    {currencies.map(currency => (
      <option key={currency} value={currency}>
        {currency}
      </option>
    ))}
  </select>
);

SelectCurrency.propTypes = {
  currencies: PropTypes.array,
  onChange: PropTypes.func,
  value: PropTypes.string
};

SelectCurrency.defaultProps = {
  currencies: ["CAD", "USD", "EUR"],
  onChange: null,
  value: ""
};

export default SelectCurrency;
