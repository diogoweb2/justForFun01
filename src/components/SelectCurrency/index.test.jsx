import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import SelectCurrency from "./index";

let wrapper;
describe("SelectCurrency", () => {
  const fn = jest.fn();

  it("should render", () => {
    wrapper = shallow(<SelectCurrency />);
    expect(wrapper).toHaveLength(1);
  });

  it("should change currency", () => {
    wrapper = shallow(
      <SelectCurrency onChange={fn} currencies={["CAD", "USD"]} />
    );
    wrapper.find("select").simulate("change", { target: { value: "USD" } });
    expect(fn).toHaveBeenCalled();
  });

  it("should match snapshot...", () => {
    wrapper = shallow(<SelectCurrency currencies={["CAD", "USD"]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
