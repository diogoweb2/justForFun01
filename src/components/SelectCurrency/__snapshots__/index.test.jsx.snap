// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectCurrency should match snapshot... 1`] = `
<div
  className="slds-select_container"
>
  <select
    className="slds-select"
    onChange={[Function]}
    value=""
  >
    <option
      key="CAD"
      value="CAD"
    >
      CAD
    </option>
    <option
      key="USD"
      value="USD"
    >
      USD
    </option>
  </select>
</div>
`;
